### ISSUE TITLE
Error starting wsl-vpnkit 0.3.8

### ISSUE THREAD

**Issue by adalsa91** [sakai135/wsl-vpnkit#191]
I get an error when tried to start release wsl-vpnkit 0.3.8, no problem with previous versions. I'm using Windows 10 22H2 (OS Build 19045.2673)

```
PS > wsl --version
WSL version: 1.1.3.0
Kernel version: 5.15.90.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.2673

PS > wsl -d wsl-vpnkit038
Processing fstab with mount -a failed.

<3>WSL (8) ERROR: CreateProcessEntryCommon:368: getpwuid(0) failed 2
<3>WSL (8) ERROR: CreateProcessEntryCommon:372: getpwuid(0) failed 2
<3>WSL (8) ERROR: CreateProcessEntryCommon:574: execvpe /bin/sh failed 2
<3>WSL (8) ERROR: CreateProcessEntryCommon:583: Create process not expected to return
```

**Comment by sakai135** [comment]
Does this happen even after a full restart of your machine?

**Comment by adalsa91** [comment]
Yes, I did. It's a fresh installation of windows and WSL is up to date so I have no idea what could be the problem.

**Comment by slatsinoglou** [comment]
Hi, I have noticed the same issue after updating to the latest version (tag) as you can see below.

❯ wsl.exe -d wsl-vpnkit service wsl-vpnkit start
<3>WSL (788) ERROR: CreateProcessParseCommon:786: Failed to translate \\wsl.localhost\Ubuntu\home\toor
❯ pwd
/home/toor


Note that wsl-vpnkit works fine after start, The path (Failed to translate) always refers to the current dir. I was just curious why this started showing up.

**Comment by sakai135** [comment]
@slatsinoglou Your error message is a different issue, #150. 

It occurs because wsl.exe tries to open the current directory in the WSL distro, which makes sense when you're in a Windows dir but not when running wsl.exe from within another WSL2 distro.

**Comment by sakai135** [comment]
@adalsa91 can you try the suggestions in https://github.com/microsoft/WSL/issues/9018 ?

**Comment by ValentinTwin1206** [comment]
Hi all, 

I did not face this issue with WSL 1.1.3 and VPNKit 0.38. Even with the latest release of WSL 1.1.6 everything seems to work.

Anyway, why isn't it recommended aynmore to add the booting of VPNKit into ~/.profile and / or ~/.bashrc of WSL2 distro?
A colleague of mine tried it with latest release of VPNKit and during booting of his WSL2 distro he faced an issue telling:
`could not find /bin/ash`... typo in any shebang??

And we were also not able to setup the VPNKit with our WSL2 distro when starting it in the foreground...

As said, the old setup with 0.38 is working fine for me.

Best Valentin

**Comment by sakai135** [comment]
Closing due to inactivity

**Comment by Baljak** [comment]
Just to mention that I couldn't get v0.4.1 to work properly (``wget example.com`` didn't work, nor did ``nslookup 1.1.1.1``), so I tried v0.3.8 which didn't work with the same error messages as @adalsa91. However, v0.3.7 works perfectly. My company's VPN is Pulse Secure 9.1.13 and a proxy is used. Previously, I've used other 0.3.x  versions without any problem (v0.3.8 is the only one that doesn't work at all).
